OurUmbraco/Project/Api/ProjectUploadController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Newtonsoft.Json;
+using OurUmbraco.Wiki.BusinessLogic;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Http;
+
+namespace OurUmbraco.Project.Api
+{
+    public class ProjectUploadController : UmbracoMemberAuthApiController
+    {
+        /// <summary>
+        /// Used by the package CLI tool, gets a list of project files to check if one already exists with the same name
+        /// </summary>
+        public IEnumerable<WikiFile> GetProjectFiles()
+        {
+            // The project/member id are exposed as base properties and are resolved from the identity created on the request
+            var files = WikiFile.CurrentFiles(ProjectNodeId);
+            return files.Where(x => x.FileType == "package" || x.FileType == "hotfix").OrderByDescending(x => x.Version.Version).ToList();
+        }
+
+        /// <summary>
+        /// Posts a package file + some related meta data to update a package from the CLI package tool.
+        /// Includes the zip, whether it should be current, the compatible .NET and Umbraco versions.
+        /// </summary>
+        /// <returns>HttpResponseMessage</returns>
+        /// <exception cref="HttpResponseException"></exception>
+        [HttpPost]
+        public async Task<HttpResponseMessage> UpdatePackage()
+        {
+            // Check if the request contains multipart/form-data.
+            if (!Request.Content.IsMimeMultipartContent())
+            {
+                throw new HttpResponseException(HttpStatusCode.UnsupportedMediaType);
+            }
+
+            string root = HttpContext.Current.Server.MapPath("~/App_Data");
```

Comment:
This should be a different directory, at least a subdirectory under `~/App_Data` since it will pollute the root folder now and I don't see files being deleted afterwards.
Second note: we need to delete these temp files when we're done with them. :-)

Reply:
Fixed in https://github.com/umbraco/OurUmbraco/pull/552/commits/4e4bc287aafab1d7c967b67d66c6ecb5a2cf2c8b

Reply:
Ah, I don't think you can delete the directory, if a few people upload files at the same time you'll get weird errors, but can just delete the temp file when the package has been created.

Reply:
Fixed in 9124934a80be4034f1e147cc6a79b24d0fcd71a4